packages/adapter-prisma/src/migrations.ts

```diff
@@ -22,21 +22,34 @@ function runMigrateWithDbUrl<T>(dbUrl: string, cb: () => T): T {
   }
 }
 
+async function withMigration<T>(
+  dbUrl: string,
+  schemaPath: string,
+  cb: (migrate: Migrate) => Promise<T>
+) {
+  await ensureDatabaseExists(dbUrl, path.dirname(schemaPath));
+  const migrate = new Migrate(schemaPath);
+
+  try {
+    return cb(migrate);
```

Comment:
You need to await the return value here otherwise the stop() is happening at the wrong time

Reply:
Nice catch, thanks!